unlock-app/src/hooks/useMultipleRecipient.ts

```diff
@@ -30,19 +30,32 @@ interface RecipientPayload {
 
 export const useMultipleRecipient = (
   lock: Lock,
-  {
-    minRecipients,
-    maxRecipients,
-    metadataInputs,
-  }: Pick<
-    PaywallConfig,
-    'metadataInputs' | 'maxRecipients' | 'minRecipients'
-  > = {
-    metadataInputs: [],
-    maxRecipients: 1,
-    minRecipients: 1,
-  }
+  paywallConfig?: PaywallConfig
 ) => {
+  let maxRecipients = 1
+  if (paywallConfig?.maxRecipients) {
+    maxRecipients = paywallConfig.maxRecipients
+  }
+  if (lock?.address && paywallConfig?.locks[lock.address]) {
+    maxRecipients = paywallConfig?.locks[lock.address].maxRecipients || 1
+  }
+
+  let minRecipients = 1
+  if (paywallConfig?.minRecipients) {
+    minRecipients = paywallConfig.minRecipients
+  }
+  if (lock?.address && paywallConfig?.locks[lock.address]) {
+    minRecipients = paywallConfig?.locks[lock.address].minRecipients || 1
+  }
+
+  let metadataInputs: Array<any> = []
+  if (paywallConfig?.metadataInputs) {
+    metadataInputs = paywallConfig.metadataInputs
```

Comment:
I would recommend using `Object.assign` or spread on paywallConfig to fill up default values for required parameters. That would be much cleaner and expandable than these if statements to track of.

```ts
Object.assign({ min: 1}, paywallConfig)
```